rtl8720dn/wioterminal.go

```diff
@@ -0,0 +1,35 @@
+//go:build wioterminal
+// +build wioterminal
+
+package rtl8720dn
+
+import (
+	"device/sam"
+	"machine"
+	"runtime/interrupt"
+)
+// UART_RTL8720DN is the WioTerminal's SPI port that is connected to the 
+// RTL8720DN device. To use it is required to reconfigure the SPI as a UART.
+var (
+	UART_RTL8720DN *machine.UART
+)
+
+// NewForWioterminal returns a new RTL8720DN driver that is ready to use
+// on the Wioterminal by reconfiguring the SPI interface to be used as a UART.
+func NewForWioterminal() *Driver {
+	UART_RTL8720DN = &machine.UART{
+		Buffer: machine.NewRingBuffer(),
+		Bus:    sam.SERCOM0_USART_INT,
+		SERCOM: 0,
+	}
+
+	UART_RTL8720DN.Interrupt = interrupt.New(sam.IRQ_SERCOM0_2, handleInterrupt)
+
+	return New(UART_RTL8720DN, machine.PB24, machine.PC24, machine.RTL8720D_CHIP_PU)
+}
+
+func handleInterrupt(interrupt.Interrupt) {
+	// should reset IRQ
+	UART_RTL8720DN.Receive(byte((UART_RTL8720DN.Bus.DATA.Get() & 0xFF)))
+	UART_RTL8720DN.Bus.INTFLAG.SetBits(sam.SERCOM_USART_INT_INTFLAG_RXC)
+}
```

Comment:
Starting with TinyGo 0.26, only the following should be required.


```go
func NewForWioterminal() *Driver {
	return New(machine.UART3, machine.PB24, machine.PC24, machine.RTL8720D_CHIP_PU)
}
```

refs: https://github.com/tinygo-org/tinygo/pull/3110

Reply:
I was able to remove a bunch of code as a result of better understanding now. Thank you for your help @sago35